pycsob/client.py

```diff
@@ -17,6 +20,202 @@ def send(self, request, **kwargs):
+    amount: int
+    description: Optional[str] = None
+    
+    def _format_name(self, value: str) -> str:
+        return value[:20]
+    
+    def _format_description(self, value: str) -> str:
+        return value[:40]
+
+
+@dataclass
+class CustomerAccount(ConvertMixin):
+    """Customer account data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customeraccount-data-
+
+    created_at: Optional[str] = None
+    changed_at: Optional[str] = None
+    changed_pwd_at: Optional[str] = None
+    order_history: Optional[int] = None
+    payments_day: Optional[int] = None
+    payments_year: Optional[int] = None
+    one_click_adds: Optional[int] = None
+    suspicious: Optional[bool] = None
+
+
+@unique
+class CustomerLoginType(Enum):
+    """Type of customer login."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    GUEST = "guest"
+    ACCOUNT = "account"
+    FEDERATED = "federated"
+    ISSUER = "issuer"
+    THIRDPARTY = "thirdparty"
+    FIDO = "fido"
+    FIDO_SIGNED = "fido_signed"
+    API = "api"
+
+
+@dataclass
+class CustomerLogin(ConvertMixin):
+    """Customer login data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    auth: Optional[CustomerLoginType] = None
+    auth_at: Optional[str] = None
+    auth_data: Optional[str] = None
+
+    def _format_auth(self, value: CustomerLoginType) -> str:
+        return value.value
+
+
+@dataclass
+class CustomerData(ConvertMixin):
+    """Customer data for creating card payment."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customer-data-
+
+    name: str
+    email: Optional[str] = None
+    home_phone: Optional[str] = None
+    work_phone: Optional[str] = None
+    mobile_phone: Optional[str] = None
+    account: Optional[CustomerAccount] = None
+    login: Optional[CustomerLogin] = None
+    
+    def _format_name(self, value: str) -> str:
+        return value[:45]
+    
+    def _format_description(self, value: str) -> str:
+        return value[:100]
+
+
+@dataclass
+class OrderAddress(ConvertMixin):
+    """Order address (billing or shipping)."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#orderaddress-data-
+
+    address_1: str
+    city: str
+    zip: str
+    country: str
+    address_2: Optional[str] = None
+    address_3: Optional[str] = None
+    state: Optional[str] = None
+    
+    def _format_address_1(self, value: str) -> str:
+        return value[:50]
+    
+    def _format_address_2(self, value: str) -> str:
+        return value[:50]
+    
+    def _format_address_3(self, value: str) -> str:
+        return value[:50]
+    
+    def _format_city(self, value: str) -> str:
+        return value[:50]
+    
+    def _format_zip(self, value: str) -> str:
+        return value[:16]
+
+    def _fields(self) -> Tuple[str, ...]:
```

Comment:
This feature is not needed. See previous comments.

Reply:
Once I dropped (as a result of your suggestion) requirements for having required fields first, this is really no more needed (and was removed).